spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientAutoConfiguration.java

```diff
@@ -80,6 +84,19 @@ RestClientBuilder elasticsearchRestClientBuilder(ElasticsearchRestClientProperti
 			return builder;
 		}
 
+		@Bean
+		@Conditional(IsSnifferAvailableOnClasspathCondition.class)
```

Comment:
That's not how classpath checks work. Can you please have a look to `@ConditionalOnClass` and how it's used in other parts of this codebase (in particular, it's not added on methods, check the Javadoc for more details).